tests/RobotFramework/libraries/ThinEdgeIO/ThinEdgeIO.py

```diff
@@ -1110,6 +1110,41 @@ def register_entity(
         json_output = json.loads(output.stdout)
         return json_output
 
+    @keyword("Get Entity")
+    def get_entity(
+            self,
+            topic_id: str,
+            device_name: str = None 
+    ) -> Dict[str, Any]:
+        """
+        Get the entity from the entity store
+
+        Args:
+            topic_id (str, optional): Topic ID of the entity
+            device_name (str, optional): Device name to fetch the entity list from
+
+        Returns:
+            Dict[str, Any]: Entity metadata
+
+        *Example:*
+        | ${entities}= | Get Entity | device/child0// |
+        | ${entities}= | Get Entity | device/child0/service/service0 | device_name=${PARENT_SN} |
+        """
+        device = self.current
+        if device_name:
+            if device_name in self.devices:
+                device = self.devices.get(device_name)
+
+        if not device:
+            raise ValueError(
+                f"Unable to query the entity store as the device: '{device_name}' has not been setup"
+            )
+
+        command = f"curl http://localhost:8000/tedge/v1/entities/{topic_id}"
```

Comment:
Using `tedge http get` will make it will ensure it also works when using cert auth for the local services.

```suggestion
        command = f"tedge http get /tedge/v1/entities/{topic_id}"
```

Reply:
I guess the same thing can apply to other similar methods ("Register Entity", "Deregister Entity", "List Entities")

Reply:
Resolved by 605ef13

Reply:
I had to revert that commit because that made the tests more flaky. It appears that the flakiness observed with that change is very similar to the flaky tests reported in #3559 . So, I am suspecting something with the `tedge http` utility that is causing this flakiness because local flake finder runs were consistently failing while using the `tedge http` utility, but they are passing once reverted back to `curl`.

Reply:
ok, we need to get to the bottom of this before the next release

Reply:
Looks like we've found the root cause of the `tedge http` vs `curl` behaviour. https://github.com/thin-edge/thin-edge.io/pull/3581

Still confirming the fix by running the system test 100 times, but 50 iterations of the test in https://github.com/thin-edge/thin-edge.io/issues/3559 resulted in 0 failures.